spec/defines/authzd_user_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+require 'faker'
+
+describe 'nebula::authzd_user' do
+  let(:title) { Faker::Internet.user_name }
+  let(:home) { '/some/where' }
+  let(:params) do
+    {
+      gid: Faker::Internet.user_name,
+      home: home,
+      key: {
+        type: 'ssh-rsa',
+        data: 'CCCCCCCCCCCC',
+        comment: Faker::Internet.email,
+      },
+    }
+  end
+
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) { os_facts }
+
+      describe 'users' do
+        it { is_expected.to contain_user(title).with(name: title, gid: params[:gid], home: home).that_requires("File[#{home}]") }
+        it { is_expected.to contain_file(home).with(ensure: 'directory', mode: '0755') }
```

Comment:
I'm curious about whether this test will continue passing if we use `managehome` instead of an explicit `file` resource. My bet is that it will, and you won't have to change the tests for that.

Reply:
It doesn't then `require('File[/var/haproxyctl]')`, but I'm OK with updating the test a little bit.